src/static/helpers/slasHelper.ts

```diff
@@ -131,16 +131,18 @@ export async function authorize(
   };
 
   const response = await slasClientCopy.authorizeCustomer(options, true);
-  const redirectUrl = response.headers?.get('location') || response.url;
+  const redirectUrlString = response.headers?.get('location') || response.url;
+  const redirectUrl = new URL(redirectUrlString);
+  const searchParams = Object.fromEntries(redirectUrl.searchParams.entries());
 
   // url is a read only property we unfortunately cannot mock out using nock
   // meaning redirectUrl will not have a falsy value for unit tests
   /* istanbul ignore next */
-  if (response.status !== 303 || !redirectUrl) {
+  if (response.status >= 400 || searchParams.error) {
```

Comment:
Hey Kevin everything looks good so I'll go ahead and approve it, but is there a specific reason you error check against the `searchParams`? Just asking for clarity!

Reply:
Hey Joel,

The redirect callback is sometimes called with the error parameter `/callback?error=invalid_request`. We need to throw an error when this happens.

Reply:
AH I apologize I just read your description more throughly, thanks for clarifying!